sdk/core/azure-core/azure/core/messaging.py

```
@@ -0,0 +1,148 @@
+    :vartype time: ~datetime.datetime
+    :ivar dataschema: Identifies the schema that data adheres to.
+    :vartype dataschema: str
+    :ivar datacontenttype: Content type of data value.
+    :vartype datacontenttype: str
+    :ivar subject: This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :vartype subject: str
+    :ivar specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :vartype specversion: str
+    :ivar id: An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :vartype id: str
+    :ivar extensions: A CloudEvent MAY include any number of additional context attributes
+     with distinct names represented as key - value pairs. Each extension must be alphanumeric, lower cased
+     and must not exceed the length of 20 characters.
+    :vartype extensions: dict
+    """
+
+    def __init__(self, source, type, **kwargs):  # pylint: disable=redefined-builtin
+        # type: (str, str, **Any) -> None
+        self.source = source
+        self.type = type
+        self.specversion = kwargs.pop("specversion", "1.0")
+        self.id = kwargs.pop("id", str(uuid.uuid4()))
+        self.time = kwargs.pop("time", datetime.now(TZ_UTC))
+        self.datacontenttype = kwargs.pop("datacontenttype", None)
+        self.dataschema = kwargs.pop("dataschema", None)
+        self.subject = kwargs.pop("subject", None)
+        self.extensions = {}
+        _extensions = dict(kwargs.pop("extensions", {}))
+        for key in _extensions.keys():
+            if not key.islower() or not key.isalnum():
+                raise ValueError(
+                    "Extension attributes should be lower cased and alphanumeric."
+                )
+        self.extensions.update(_extensions)
+        self.data = kwargs.pop("data", None)
```

I would expect that we raise an error for unexpected (remaining) `kwargs`.

---

i am not sure it's a good idea - wouldn't this prevent us from having additive changes in the future?

---

How would this prevent us from having additive changes in the future?

---

hmm, i was thinking on the lines of "Aren't we blocking  adding keyword args in the future", but i got it wrong - updated